pkg/kubelet/kuberuntime/fake_kuberuntime.go

```diff
@@ -0,0 +1,480 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+var (
+	version                 = "0.1.0"
+	fakeImageSize    uint64 = 1
+	fakeRuntimeName         = "fakeRuntime"
+	fakePodSandboxIP        = "192.168.192.168"
+)
+
+type PodSandboxWithState struct {
+	// creation timestamp
+	createdAt int64
+	// the config of this pod sandbox
+	config *runtimeApi.PodSandboxConfig
+	// the state of this pod sandbox
+	state runtimeApi.PodSandBoxState
+}
+
+type ContainerWithState struct {
+	// creation timestamp
+	createdAt int64
+	// the id of this container
+	containerID string
+	// the sandbox id of this container
+	podSandboxID string
+	// the id of the image
+	imageID string
+	// the state of this container
+	state runtimeApi.ContainerState
+	// the config of this container
+	containerConfig *runtimeApi.ContainerConfig
+	// the sandbox config of this container
+	sandboxConfig *runtimeApi.PodSandboxConfig
+}
+
+type fakeKubeRuntime struct {
+	sync.Mutex
+
+	Images     map[string]*runtimeApi.Image
+	Containers map[string]*ContainerWithState
+	Sandboxes  map[string]*PodSandboxWithState
+
+	Called []string
+}
+
+func NewFakeKubeRuntime() *fakeKubeRuntime {
+	s := &fakeKubeRuntime{
+		Images:     make(map[string]*runtimeApi.Image),
+		Containers: make(map[string]*ContainerWithState),
+		Sandboxes:  make(map[string]*PodSandboxWithState),
+	}
+
+	return s
+}
+
+func stringInSlice(in string, list []string) bool {
```

Comment:
nit: rename the variable `in` to `s` or `str`.
This seems like a general helper function. If we don't have this already, just add it to a new package under `pkg/kubelet/util`